simulation/sub8_gazebo/nodes/fake_perception.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/python
+#TODO
+#Get modularity working
+#Generate Services using a for loop
+#Step 1: Parse Launch File into this so I can read new models and coords directly from there
+#Step 2: Figure out which service is accessing master pose function
+#Step 3: Making master junk functions
+#Step 4: Make the whole thing a class <<< This should come first.
+import cv2
+from geometry_msgs.msg import Pose, Point, Quaternion, PoseStamped, Vector3Stamped, Vector3
+from mil_misc_tools import text_effects
+from mil_msgs.srv import SetGeometry
+import numpy as np
+import rospy
+from std_srvs.srv import SetBool, SetBoolResponse
+from sub8_msgs.srv import VisionRequestResponse, VisionRequest, VisionRequest2D, VisionRequest2DResponse
+from std_msgs.msg import Header, Float64
+import os
+import rospkg
+
+import sys
+import time
+from tf import transformations
+import yaml
+import xml.etree.ElementTree
+rospack = rospkg.RosPack()
+#Find a way to parse through the yaml correctly. Might be unneccessary but nice.
+config_file = os.path.join(rospack.get_path('sub8_missions'), 'sub8', 'vision_proxies.yaml')
+f = yaml.load(open(config_file, 'r'))
+launch_file = os.path.join(rospack.get_path('sub8_gazebo'), 'launch', 'duck.launch')
+launch = xml.etree.ElementTree.parse(launch_file).getroot()
+
+fprint = text_effects.FprintFactory(title="SIMULATOR").fprint
+
+def handle_fake_perception(extra,resp):
+    now = rospy.get_rostime()
+    k = np.uint32(0)
+    if extra != '':
+        resp = extra
+    if resp == '':
+        print "NO TARGET"
+        sys.exit(0)
+    for node in launch.iter('node'):
+        name = node.get('name')
+        if name == resp:
+            v = node.get('args')
+            point = getCoords(resp,v)
+            q = getOrient(resp,v)
+
+    pose_stamp = PoseStamped(header = Header(seq = k, stamp = now, frame_id = "/map"),
+        pose = Pose(position=point,
+        orientation=Quaternion(*q)))
+    covariance_diagonal = Vector3(0,0,0)
+    found = True
+    resp2 = VisionRequestResponse(pose_stamp,covariance_diagonal,found)
+    return resp2
```

Comment:
So we're parsing the entire XML file every single service request? This data doesn't change, so it would be better to parse the file at the start of the program, store it somewhere, and return this cached pose in the service response.